iommi/live_edit.py

```diff
@@ -233,15 +329,175 @@ def dangerous_execute_code(code, request, view):
+                class Link:
+                    def get_absolute_url(self):
+                        return '#'
+
+                    def __str__(self):
+                        return 'title'
+
+                return Link()
+            if shortcut == 'number':
+                return f'{self.idx}'
+            return s
+
+        @staticmethod
+        def get_absolute_url():
+            return '#'
+
+    return Page(
+        iommi_style=style,
+        parts=dict(
+            title=Header('Style showcase'),
+            menu__children=dict(
+                menu_title=html.h2('Menu'),
+                menu=Menu(
+                    sub_menu=dict(
+                        active=MenuItem(url=request.get_full_path()),  # full path do make this item active
+                        inactive=MenuItem(),
+                    )
+                ),
+            ),
+            form=Form(
+                title='Form',
+                fields=dict(
+                    text=Field.text(initial='initial'),
+                    boolean=Field.boolean(),
+                    boolean_selected=Field.boolean(initial=True),
+                    radio=Field.radio(choices=['a', 'b', 'c'], initial='b'),
+                ),
+                actions__submit__post_handler=lambda **_: None,
+                actions__secondary=Action.button(),
+                actions__delete=Action.delete(display_name='Delete'),
+                actions__icon=Action.icon('trash', display_name='Icon', attrs__href='#'),
+            ),
+            table=Table(
+                title='Table',
+                assets__ajax_enhance__template=None,
+                model=User,
+                columns={
+                    t.__name__: dict(call_target=t, display_name=t.__name__)
+                    for t in [Column.select, Column.edit, Column.delete, Column.boolean, Column.text, Column.number, Column.link, Column.icon]
+                },
+                columns__text__filter__include=True,
+                columns__number__filter__include=True,
+                rows=[DummyRow(i) for i in range(10)],
+                page_size=2,
+            ),
+        ),
+    )
+
+
+def style_editor__select(**_):
+    from iommi.style import _styles
+    return Form(
+        title='Select style to edit',
+        fields__name=Field.choice(
+            choices=[
+                k
+                for k, v in items(_styles)
+                if not v.internal
+            ]
+        ),
+        actions__edit=Action.primary(
+            display_name='Edit',
+            post_handler=lambda form, **_: HttpResponseRedirect(f'?_iommi_live_edit=style_editor__edit&name={form.fields.name.value}') if form.is_valid() else None,
+        ),
+        actions__new_style__attrs__href='?_iommi_live_edit=style_editor__new',
+    )
+
+
+@csrf_exempt
+def style_editor__edit(request, **_):
+    from iommi.style import get_style, register_style
+    name = request.GET.get('name', request.POST['name'])
```

Comment:
This line fails in the GET case. I think it should be:
```python
   name = request.GET.get('name', request.POST.get('name'))
```

Reply:
Doh